addons/edi/models/edi_synchronizer.py

```diff
@@ -488,6 +501,23 @@ def log_progress(self, doc, log_message=None, total_records=None, records_proces
             _logger.info(log_message)
         doc.create_or_update_edi_progress(log_message, total_records, records_processed)
 
+    def set_error_message(self, error_message, log_mode):
+        """
+        Error message which is caught in exception from ValidationError or IntegrityError
+        """
+        towrite = self.env.all.towrite.copy()
+        tocompute = self.env.all.tocompute.copy()
+        # TODO: use `with api.Environment.manage():` instead of towrite/tocompute manipulation
```

Comment:
Remove this comment, I have tried it and it doesn't work. Saving and restoring `towrite` & `tocompute` is still required.